src/main/java/org/reaktivity/nukleus/kafka/internal/cache/CompactedPartitionIndex.java

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.kafka.internal.cache;
+
+import static org.reaktivity.nukleus.kafka.internal.util.BufferUtil.EMPTY_BYTE_ARRAY;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.agrona.DirectBuffer;
+import org.agrona.collections.LongArrayList;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.nukleus.kafka.internal.stream.HeadersFW;
+import org.reaktivity.nukleus.kafka.internal.types.MessageFW;
+import org.reaktivity.nukleus.kafka.internal.types.OctetsFW;
+
+public class PartitionIndex implements IPartitionIndex
```

Comment:
Class name does not match source name.
Suggest `DefaultPartitionIndex` if interface is also still needed.

Reply:
That was an intermediate state, fixed now